src/Tests/Tests.Reproduce/GithubIssue3907.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Net;
+using Elastic.Xunit.XunitPlumbing;
+using FluentAssertions;
+using Nest;
+using Tests.Core.ManagedElasticsearch.Clusters;
+using Tests.Domain;
+
+namespace Tests.Reproduce
+{
+	public class GithubIssue3907 : IClusterFixture<ReadOnlyCluster>
+	{
+		private readonly IntrusiveOperationCluster _cluster;
+
+		// use intrusive operation because we're changing the underlying http handler
+		public GithubIssue3907(IntrusiveOperationCluster cluster) => _cluster = cluster;
+
+		[I]
+		public void NotUsingSocketsHttpHandlerDoesNotCauseException()
+		{
+			AppContext.SetSwitch("System.Net.Http.UseSocketsHttpHandler", false);
```

Comment:
Add comment this is safe to do because `IntrusiveOperationCluster` runs with max concurrency of 1